src/view.js

```diff
@@ -47,6 +47,21 @@ module.exports = function (Vue) {
         return this.invalidate()
       }
 
+      // for every route run global before function
```

Comment:
I think the global before should not be triggered here - this function is called for very nested view on a route change. The global before should be called once only for each route change, so it should actually be in the `_match` method of the router class.